packages/state-transition/src/cache/stateCache.ts

```diff
@@ -162,9 +162,9 @@ export function createCachedBeaconState<T extends BeaconStateAllForks>(
  * Create a CachedBeaconState given a cached seed state and state bytes
  * This guarantees that the returned state shares the same tree with the seed state
  * Check loadState() api for more details
- * // TODO: rename to loadUnfinalizedCachedBeaconState() due to ELECTRA
+ * Note: This only loads unfinalized beacon state
  */
-export function loadCachedBeaconState<T extends BeaconStateAllForks & BeaconStateCache>(
+export function loadUnfinalizedCachedBeaconState<T extends BeaconStateAllForks & BeaconStateCache>(
```

Comment:
this should stay the original name since the state might not have to do anything with finalized/unfinalized pubkey indices

Reply:
> this should stay the original name since the state might not have to do anything with finalized/unfinalized pubkey indices

I see. Yea I agree the `unfinalized` here is unrelated but rather it's part of the persistent state cache. Will rename it in the next PR.

Reply:
@ensi321 what differences between loading a finalized state vs unfinalized state?

Reply:
> @ensi321 what differences between loading a finalized state vs unfinalized state?

@tuyennhv In the context of 6110? It probably doesn't make a difference.

> TODO: rename to loadUnfinalizedCachedBeaconState() due to EIP-6110
When you first added this comment in this [PR](https://github.com/ChainSafe/lodestar/pull/6176/files) I thought you knew something that I don't 😅

Reply:
just review the logic, right now when loading any states we use `EpochCache.createFromState` which set empty `unfinalizedPubkey2index` which is not correct for unfinalized states

in this function, we should pass in `pivotValidatorIndex` to decide `unfinalizedPubkey2index`, hence the name should be `loadUnfinalizedCachedBeaconState`